flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -160,12 +168,13 @@ public void setCustomPartitioner(KinesisPartitioner<OUT> partitioner) {
 	public void open(Configuration parameters) throws Exception {
 		super.open(parameters);
 
-		KinesisProducerConfiguration config = new KinesisProducerConfiguration();
-		config.setRegion(this.region);
-		config.setCredentialsProvider(new StaticCredentialsProvider(new BasicAWSCredentials(this.accessKey, this.secretKey)));
+		KinesisProducerConfiguration producerConfig = new KinesisProducerConfiguration();
+
+		producerConfig.setRegion(configProps.getProperty(KinesisConfigConstants.CONFIG_AWS_REGION));
+		producerConfig.setCredentialsProvider(AWSUtil.getCredentialsProvider(configProps));
 		//config.setCollectionMaxCount(1);
 		//config.setAggregationMaxCount(1);
```

Comment:
Sorry again for my late response. Meanwhile I have thought about this and I have come up with a question. I had noticed a remark about not using the KCL library, but is there any special reason why we are not using KPL? Anything we intend to do will probably replicate it including retry mechanism on failures, multi-threading and asynchronous writes.

We also need to take into consideration that our producer solution should be cost effective by using the opportunities for batching, which can be customized by the user if low latency is preferred.